### PR TITLE
add to pypi

### PR DESCRIPTION
**Pull request by johanneswilm**
Hey,
given that there was no action on the part of @pradyunsg over the past two years [1] and we badly needed this, I have published the python package to pypi [2]. In order to fulfill the pypi maintenance requirements, I really needed write access, so I did that out of our fork for now. However, I really don't have any shares in this package, so if @pradyunsg, @tobeycarman, @ronichoudhury or @getify want to take over package maintainership, just let me know and I will happily transfer it to you.

[1] https://github.com/getify/JSON.minify/issues/21
[2] https://pypi.python.org/pypi/JSON_minify

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,28 +1,26 @@
-from setuptools import setup
+from setuptools import setup, find_packages
 
-try:
```
Comment (pradyunsg) [inline setup.py]: Let this try/except stay?

File: setup.py [hunk 2]
```diff
@@ -1,28 +1,26 @@
...
     version="0.2.0",
+    include_package_data=True,
+    license='MIT License',
     description=(
-        "A simple script to minify valid JSON, containing C/C++ style comments"
+        "A simple script to minify valid JSON, containing JavaScript comments"
     ),
     long_description=long_description,
-    url="https://github.com/getify/JSON.minify/tree/python",
+    url="https://github.com/fiduswriter/JSON.minify/tree/python",
```
Comment (pradyunsg) [inline setup.py]: You'll want to revert this line change before this merges. :P

File: setup.py (hunk 1 above)
Comment (johanneswilm) [reply setup.py]: What is the point of the try/except?

**Comment by pradyunsg** [comment]
Sorry; I completely lost track of this. Thanks for doing this @johanneswilm!

If you want it and @getify is fine with it, he can give you write access to the repo and you can maintain the package. Otherwise, I don't mind taking over the name and managing it. :)

File: setup.py (hunk 2 above)
Comment (johanneswilm) [reply setup.py]: right. if you guys are willing to take over maintainership, we cna move it all back into the getify repository.

**Comment by johanneswilm** [comment]
@pradyunsg Hey, yes, I would prefer for you to take over maintainership (I have too many other obligations already). If you give me your username on pypi, I can make the transfer.

**Comment by pradyunsg** [comment]
It's `pradyunsg`. :)

File: setup.py (hunk 1 above)
Comment (pradyunsg) [reply setup.py]: Messy permissions on some system can cause this to fail in hilarious ways. It doesn't matter much though.

I'm gonna switch this package to using `flit` for packaging if I get the time.

**Comment by johanneswilm** [comment]
@pradyunsg Ok, great, I have added you as a package owner. Feel free to remove me once you have everything set up.

**Comment by pradyunsg** [comment]
I'll do it in a few hours. :)

File: setup.py (hunk 1 above)
Comment (johanneswilm) [reply setup.py]: I think you need to fix your permissions. Writing crappy software because someone doesn't have their system set up rioght is usually not a good idea. But given that you are the maintainer here, you are welcome to make the change yourself if you really think it is a good idea.

**Comment by pradyunsg** [comment]
Closing due to #45 and https://pypi.org/project/JSON_minify/ .

**Comment by pradyunsg** [comment]
Sorry for the delay. Got swept in other stuff.

**Comment by pradyunsg** [comment]
@johanneswilm Could you look at https://pypi.org/manage/project/JSON_minify/collaboration/ and verify if I'm a project owner?

**Comment by johanneswilm** [comment]
@pradyunsg you are a manager so that you can push out releases. I stay myself also just to ensure that releases really do happen given that there has been a history of updates not coming through.

**Comment by pradyunsg** [comment]
Mind making me an owner? =)

**Comment by johanneswilm** [comment]
I actually made you an owner first. But then when you didn't release for another 3 weeks even though the code was ready and had written in that there was no pypi package, I took that position again. Given that you apparently don't use the package on pypi yourself, I think this current setup is quite OK, don't you think? We are a bunch of different projects who need a package like this and had been waiting for years for a similar package to finally update to python 3, and the administrator just stopped reacting. So now it would be good to avoid having the same thing happen here.

**Comment by pradyunsg** [comment]
> when you didn't release for another 3 weeks

I do this in my own free time. I did it when I found the free time to do so.

> I think this current setup is quite OK, don't you think?

No, I don't. Anyway, if you want to stay on top of everything, you can just stay an owner. That said, I don't expect there to be any activity happening on this project and at the end of the day I don't really care enough to use my time to spend time elaborating the reasons for why I don't think so.

I'm happy to let you take over maintenance of the PyPI project and remove me as a collaborator.

**Comment by johanneswilm** [comment]
> I do this in my own free time. I did it when I found the free time to do so.

Yes, and that's fine. But then you also surely understand that those who rely on this for something serious actually need to be able to make sure that the project doesn't just die on pypi, right?

> I don't really care enough to use my time to spend time

Right. So that's another good reason why you probably shouldn't take full control of the package. You are welcome to continue as a manager though.

> I'm happy to let you take over maintenance of the PyPI project and remove me as a collaborator.

Ok, I can continue to maintain it there as I have been doing, Let me know if you want to be added again.

**Comment by pradyunsg** [comment]
> something serious actually need to be able to make sure that the project doesn't just die on pypi

Nope. It's easy enough to fork and maintain. I'd rather not digress though. :)

> You are welcome to continue as a manager

I'll be happy to step down.

> I can continue to maintain it there as I have been doing

👍

**Comment by johanneswilm** [comment]
> Nope. It's easy enough to fork and maintain. I'd rather not digress though. :)

Right. But pypi maintainership is gone if the maintainer there is just unresponsive. Taht's what happened with the maintainer of a similar package. As for this package on pypi, the history is this:

1. November 2014 requested ti to be packaged. Half a year later you said it would happen. Another half a year later you said you'd do it after your exams. We are now in December 2015. Two years later, fall 2017, I fixed the package and uploaded it to pypi and offered to you to let you take it over. You didn't take my fixes and instead wrote on the github page that the package is not on pypi (when it really was). When I intervened again, you finally uploaded your latest update to pypi another few weeks later.

So basically, I am a bit afraid of you just disappearing for a few years again. That's of course your right to do -- as you sid, you do this in your free time. It's just not acceptable for a production level project to depend on a maintainer who doesn't respond.That's why I will continue to have some control over the pypi package. I don;t understand why you need to be registered as package owner there when you don't really care enough to keep an updated package there.

**Comment by pradyunsg** [comment]
> I don;t understand why you need to be registered as package owner there when you don't really care enough to keep an updated package there.

I don't. :)

**Comment by getify** [comment]
OK, seems like everything has been resolved. Locking this conversation so we don't digress any further.
